Mathlib/RingTheory/LocalProperties/Exactness.lean

```diff
@@ -0,0 +1,245 @@
+/-
+Copyright (c) 2024 Sihan Su. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sihan Su, Yongle Hu, Yi Song
+-/
+import Mathlib.RingTheory.LocalProperties.Submodule
+import Mathlib.Algebra.Exact
+import Mathlib.RingTheory.Localization.Away.Basic
+
+/-!
+# Local properties about linear maps
+
+In this file, we show that
+injectivity, surjectivity, bijectivity and exactness of linear maps are local properties.
+More precisely, we show that these can be checked at maximal ideals and on standard covers.
+
+-/
+open Submodule LocalizedModule Ideal LinearMap
+
+section localization_maximal
+
+variable {R M N : Type*} [CommRing R] [AddCommGroup M] [Module R M] [AddCommGroup N] [Module R N]
+  (f : M →ₗ[R] N)
+
+theorem injective_of_localization
+    (h : ∀ (J : Ideal R) (_ : J.IsMaximal), Function.Injective (map J.primeCompl f)) :
+    Function.Injective f :=
+  ker_eq_bot.mp <| eq_bot_of_localization_maximal' <|
+  fun J hJ ↦ (localized'_ker_eq_ker_localizedMap _ _ _ _ f).trans <| ker_eq_bot.mpr <| h J hJ
```

Comment:
Once #19118 is merged, this (and other lemmas below) can be generalized to AddCommMonoid/CommSemiring. `ker_eq_bot` doesn't work in that setting but you can directly use [Module.eq_of_localization_maximal](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/LocalProperties/Submodule.html#Module.eq_of_localization_maximal).